### PR TITLE
Optimize styles variables for theming

### PR DESCRIPTION
**Pull request by bramkragten**
Retry:

Reduced the colors, so it is easier to theme. A few colors are minimally changed.

### CODE REVIEW

**Comment by balloob** [comment]
You have a merge conflict. Make sure that when you fork off the master branch of your repo, that your master branch is up to date with the main repo. 

(or just delete your fork and fork again 👍 )

**Comment by bramkragten** [comment]
🎉

File: src/resources/ha-style.html [hunk 1]
```diff
@@ -5,53 +5,37 @@
...
-    --paper-slider-pin-color: var(--primary-color);
-    --paper-slider-active-color: var(--primary-color);
-    --paper-slider-secondary-color: var(--light-primary-color);
-    --paper-slider-container-color: var(--divider-color);
+    --paper-item-icon-color: #44739e;
+    --paper-item-icon-active-color: #FDD835;
 
-    --paper-card-background-color: #FFF;
-    --paper-listbox-background-color: #FFF;
+    --sidebar-text_-_background: var(--tertiary-background-color);
```
Comment (balloob) [inline src/resources/ha-style.html]: We should not use underscores.

File: src/resources/ha-style.html [hunk 2]
```diff
@@ -5,53 +5,37 @@
...
+
+    --secondary-background-color: #E5E5E5; /* states, */
     --secondary-text-color: #727272;
+
+    --tertiary-background-color: #FFFFFF; /* badges, state cards, side menu */
+    --tertiary-text-color: rgb(76, 76, 76); /* badges, */
+
     --disabled-text-color: #bdbdbd;
     --divider-color: rgba(0, 0, 0, .12);
+    --accent-color: #FF9800;
```
Comment (balloob) [inline src/resources/ha-style.html]: Moving variables around while not changing values makes the PR very difficult to parse. Is it a change or just a move? Please avoid it.

File: src/resources/ha-style.html [hunk 3]
```diff
@@ -5,53 +5,37 @@
...
-    --primary-background-color: var(--paper-grey-50);
-    --secondary-background-color: #E5E5E5;
+    --text-primary-color: #FFFFFF; /* header */
+    --primary-background-color: #fafafa;
     --primary-text-color: #212121;
+
+    --secondary-background-color: #E5E5E5; /* states, */
     --secondary-text-color: #727272;
+
+    --tertiary-background-color: #FFFFFF; /* badges, state cards, side menu */
```
Comment (balloob) [inline src/resources/ha-style.html]: This is exactly why I don't like theming. It becomes messy really really fast.

I don't like tertiary background, I don't want to make it an "official" color. Let's just stick to individual variables for background badges, state cards and sidebar.

The original set of variables is based on what is used by the paper-* components by Google and what the material design themes prescribe https://www.materialpalette.com/blue/orange

File: src/resources/ha-style.html (hunk 1 above)
Comment (bramkragten) [inline src/resources/ha-style.html]: It is a variable that was already used, I make --tertiary-background-color, so users can specify that and don't have to bother about the --sidebar-text_-_background. Do you want me to replace all the uses of --sidebar-text_-_background in the html with --tertiary-background-color?

File: src/resources/ha-style.html (hunk 3 above)
Comment (bramkragten) [inline src/resources/ha-style.html]: I don't like it either, I rather have less colors then we have now, but we don't. This is just a way to make it easier to theme, you can still set the individual variables, they are still there. 
But if you don't like it, we can forget it. It won't change functionality.

File: src/resources/ha-style.html [hunk 4]
```diff
@@ -5,53 +5,37 @@
...
-    /* for label-badge */
-    --label-badge-background-color: white;
-    --label-badge-text-color: rgb(76, 76, 76);
-    --label-badge-red: #DF4C1E;
-    --label-badge-blue: #039be5;
-    --label-badge-green: #0DA035;
-    --label-badge-yellow: #f4b400;
-    --label-badge-grey: var(--paper-grey-500);
+    /* for paper-toggle-button */
+    --paper-grey-50: var(--primary-background-color);
```
Comment (balloob) [inline src/resources/ha-style.html]: We should not redefine paper-greys. That's an official Material Design color.

When defining styling, there is a hierarchy to things.

1. Define color names (ie paper-grey-50)
2. Define brand colors using color names (ie primary color, accent color)
3. Define specific applications of the brand colors to UI elements using brand colors. Never use color names directly. (--paper-slider-container-color)

If we follow this correctly, theming can focus on the variables defined in layer 2.

Comment (bramkragten) [inline src/resources/ha-style.html]: OK, agree.

**Comment by balloob** [comment]
Any update on this @bramkragten ?

**Comment by bramkragten** [comment]
No, didnt have time to look in to it yet, we can close this for now I think. Will start over when I have time.

**Comment by balloob** [comment]
Alright.

**Pull request closed without merging**